### DISCUSSION TITLE
RabbitMQ and long running jobs

### DISCUSSION THREAD

**Discussion by Felipe379** [rabbitmq/rabbitmq-server#9361] [Questions]
I've been using RabbitMQ for Rebus in a .NET project and I got an issue when there's a long running job in a Saga.
Enabling logging for console shows the output:

```
[WRN] Rebus.RabbitMq.RabbitMqTransport (.NET ThreadPool Worker): Closed channel detected - consumer will be disposed
[WRN] Rebus.RabbitMq.RabbitMqTransport (.NET ThreadPool Worker): Closed channel detected - consumer will be disposed
[WRN] Rebus.RabbitMq.RabbitMqTransport (.NET ThreadPool Worker): Closed channel detected - consumer will be disposed
[WRN] Rebus.RabbitMq.RabbitMqTransport (.NET ThreadPool Worker): Closed channel detected - consumer will be disposed
[INF] Rebus.RabbitMq.RabbitMqTransport (Rebus 1 worker 1): Successfully initialized consumer for "Development_localhost"
[DBG] Rebus.Sagas.LoadSagaDataStep (.NET ThreadPool Worker): Found existing saga data with ID 07158f1d-34f7-4a7e-bb67-85b379a7b272 for message "LongRunningJob/141c1713-20a1-4d19-a1c2-b1991e05dae7"
```

Correct if I'm wrong but it seems to be some kind of timeout when there is a long running job and I noticed it happens around every 30 minutes.

This timeout causes it to restart the current job in the Saga and the job starts all over again. It would get in an infinite loop, except that in this case it throws an exception because the data processed in the previous iteration has already been saved in the database and so it throws an exception.

I would like to, if possible, increase the time that it takes to timeout. If that's not possible, is there any way to fix this current issue?

**Comment by lukebakken** [comment]
You are running into consumer timeouts - https://www.rabbitmq.com/consumers.html#acknowledgement-timeout

You can increase the timeout globally or on a per-queue basis. Please see the docs.
